packages/utils-dom/src/htmlelement.ts

```diff
@@ -217,6 +237,8 @@ export class HTMLElement extends Node {
         return {
             column: location ? location.startCol : -1,
             elementId: this._element.id,
+            endColumn: location ? location.endCol - 1 : -1,
+            endLine: location ? location.endLine - 1 : -1,
```

Comment:
Looks like I can't add an end position to elements without unintended side-effects. Here's an example of a report on the `<html>` element. With this change we now underline the entirety of the tag and all its contents:
![image](https://user-images.githubusercontent.com/785009/174126735-29c2a9a5-1067-45b5-85ad-8269c323ed77.png)

The reason - our VS Code extension looks for an omitted `endColumn` before applying logic to underline only the tag name:
https://github.com/webhintio/hint/blob/8162e2a7a04d31bb7e79e09b1e0b9a2b9365148a/packages/extension-vscode/src/utils/problems.ts#L33

This seems reasonable for underlining, but different than what we want if we need to apply an edit to replace the entire element. I may need to introduce a separate method to get the full tag range and keep this one as just the position of the tag name.

Reply:
After further review, I realized this is really only an issue for derived locations within `context.report`. So instead of splitting into separate APIs, I kept the change to include the full range in `getLocation` and `getAttributeLocation`, then updated `context.report` to use only the start of the location when extracting from a provided element/attribute.

This keeps the underlying locations more consistent with how we're using them elsewhere. For example, the `endOffset` was already present and used by `hint-meta-charset-utf8` and CSS-related locations already included `endLine` and `endColumn`.

The end result (one issue on the `img` for a missing title and another on `src` for the protocol-relative-url):
![image](https://user-images.githubusercontent.com/785009/174156304-cff8b900-e207-4a94-9b23-0611894e2ff0.png)